javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -968,7 +968,8 @@ public T getNext() {
     static <T> Iterator<T> iterate(T seed, Function<? super T, ? extends T> f) {
         Objects.requireNonNull(f, "f is null");
         return new AbstractIterator<T>() {
-            T next = seed;
+            Function<? super T, ? extends T> nextFunc = (s) -> seed;
```

Comment:
I changed the name `prev` to `next` because we recently started to implement navigable collections having a reverse Iterator. The name `prev` is therefore too confusing.